docs/tutorial.md

```diff
@@ -608,118 +687,144 @@ curl "data-store.marathon.mesos:<port>/v1/plan/status"
+automatically forwarded to the environments of the matching tasks
 as environment variables, with the `TASKCFG_<TASK_TYPE>_` prefixes
 removed. A special prefix of `TASKCFG_ALL_<NAME>` may be used for
 any options that should be passed to *every* task type.
 
-A common need for service developers is an easy way to write
-configuration files before launching tasks. To fulfill this need,
-the developer may provide configuration file template(s) in their
-TaskSet(s). These templates follow the [mustache](https://mustache.github.io/)
+You can provide configuration file template(s) in your
+`TaskSet`(s), an easy way to write configuration files before
+launching tasks. These templates follow the
+[mustache](https://mustache.github.io/)
 templating format, similar to what's used in DC/OS packaging. The
 templates will be automatically rendered against the task's
-environment (which is customized as described above), and then
-each written to relative file paths specified by the developer.
+environment (which is customized as described above) and then
+each template is written to relative file paths specified by the
+developer.
 
-## Upgrade support
-From time to time, services must be upgraded from version N to version N+1.
+## Interactive upgrade support
 Tasks must often be updated one at a time, and often depend on certain
-administrative tasks like backup/restore.  An administrator may wish to pause,
+administrative tasks like backup/restore.  You can choose to pause,
 continue, or abort an upgrade interactively.
 
 ## Fault tolerance
 There are two component failures that any scheduler must defend
-against: the scheduler and the service.
-
-The scheduler and the Mesos master each store their own state for the
-schedulers' tasks.  Assuming no failure, these states eventually
-become consistent.  However, if the scheduler fails, it must perform
-task reconciliation to reconcile its own state with that of master.
-Further, it must maintain a WAL of task states to a highly available
-durable store.
-
-If the service fails, the scheduler must detect this and restart the task on
-the same node so that it can use the existing persistent
-volume.  If the node fails to come up for a certain amount of time,
-the scheduler may need to restart the task on a different node.
-
-## Administrative tasks
-Services can require any number of administrative tasks.  A common
-task is backup/restore.  The service must allocate resources on the
-executor to perform this task, restart it upon failure, and coordinate
-it across all executors.
+against: the scheduler and the service. The SDK sets up fault
+tolerance for your scheduler automatically.
+
+The SDK abstracts away task reconciliation with the Mesos master in
+the case of task failure. In the case of service failure, the SDK
+configures your scheduler to restart the task on the same node or, if
+the node has failed, restarts the task on a different node. 
```

Comment:
We should emphasize that the persistent volume is reused on a restart, and a new one is created when the node fails.